utils/hct/hctdb.py

```diff
@@ -5510,8 +5510,509 @@ def UFI(name, **mappings):
 
         # End of DXIL 1.8 opcodes.
         self.set_op_count_for_version(1, 8, next_op_idx)
-        assert next_op_idx == 258, (
-            "258 is expected next operation index but encountered %d and thus opcodes are broken"
+
+        self.add_dxil_op(
+            "AllocateRayQuery2",
+            next_op_idx,
+            "Reserved",
+            "Reserved",
+            "v",
+            "",
+            [retvoid_param],
+        )
+        next_op_idx += 1
```

Comment:
This is fine, but could also be:
```suggestion
        self.add_dxil_op_reserved("AllocateRayQuery2", next_op_idx)
        next_op_idx += 1
```

Reply:
That's cleaner. Added the change in [dc7917e](https://github.com/microsoft/DirectXShaderCompiler/pull/7125/commits/dc7917e9cb5e638d846d1fa40da6779ee9b1b955)